src/user.rs

```diff
@@ -13,10 +20,68 @@ use utils;
 /// 2. Add the user's name and public key to the protonfile
 ///
 /// Impure.
-pub fn new_user<P: AsRef<Path>>(public_key_path: P, name: String) -> Result<(), Error> {
-    let pub_key = try!(utils::file_as_string(public_key_path));
+pub fn new_user<P: AsRef<Path>>(public_key_path: P, name: &str) -> Result<(), Error> {
+    let pub_key = try!(get_public_key(public_key_path));
     let project = try!(utils::read_protonfile(None::<P>));
-    let new_project = try!(project.add_user(&name, &pub_key));
+    let new_project = try!(project.add_user(name, &pub_key));
     utils::write_protonfile(&new_project, None::<P>)
 }
 
+/// Identifies a user by their private SSH key by finding the
+/// corresponding public key in the project. This private key
+/// acts like the user's password, and should be protected.
+/// 
+/// Impure.
+pub fn id_user<P: AsRef<Path>>(private_key_path: P) -> Result<User, Error> {
+    let test_data: &[u8] = b"Testing to find private/public key pair";
+    
+    let mut private_key_file = try!(File::open(&private_key_path).map_err(Error::Io));
+    
+    let private_key = try!(openssl_RSA::private_key_from_pem(&mut private_key_file)
+        .map_err(ssl_error));
+
+    let signature = try!(private_key.sign(openssl_HashType::MD5, &test_data)
+        .map_err(ssl_error));
+
+    let project = try!(utils::read_protonfile(None::<P>));
+    for user in project.users {
+        let user_key = user.public_key.clone();
+        let mut pub_key_readable = Cursor::new(&user_key);
+
+        let rsa_public = try!(openssl_RSA::public_key_from_pem(&mut pub_key_readable)
+            .map_err(Error::Ssl));
+        
+        match rsa_public.verify(openssl_HashType::MD5, &test_data, &signature) {
+            Ok(valid) => if valid {
+                return Ok(user)
+            },
+            Err(e) => return Err(ssl_error(e)),
+        };
+    };
+    
+
+    let private_key_str = try!(utils::file_as_string(&private_key_path));
+    Err(user_not_found_error(private_key_str))
+}
+
+fn get_public_key<P: AsRef<Path>>(public_key_path: P) -> Result<String, Error> {
+    let pub_key = try!(utils::file_as_string(public_key_path));
+    let mut pub_key_readable = Cursor::new(&pub_key);
+    match openssl_RSA::public_key_from_pem(&mut pub_key_readable) {
+        Ok(_) => Ok(pub_key.clone()),
```

Comment:
Yup, it does. Borrowed in line 69. Could return as a reference (&str), but then I have to tie it's lifetime to public_key_path, plus the function name indicates an ownership of the returned value. Couldn't find a better way to do this.